torchani/cuaev/cuaev.cpp

```diff
@@ -0,0 +1,198 @@
+#include <aev.h>
+#include <torch/extension.h>
+using torch::Tensor;
+using torch::autograd::AutogradContext;
+using torch::autograd::tensor_list;
+
+AEVScalarParams::AEVScalarParams(
+    float Rcr_,
+    float Rca_,
+    Tensor EtaR_t_,
+    Tensor ShfR_t_,
+    Tensor EtaA_t_,
+    Tensor Zeta_t_,
+    Tensor ShfA_t_,
+    Tensor ShfZ_t_,
+    int num_species_)
+    : Rcr(Rcr_),
+      Rca(Rca_),
+      radial_sublength(EtaR_t_.size(0) * ShfR_t_.size(0)),
+      angular_sublength(EtaA_t_.size(0) * Zeta_t_.size(0) * ShfA_t_.size(0) * ShfZ_t_.size(0)),
+      num_species(num_species_),
+      EtaR_t(EtaR_t_),
+      ShfR_t(ShfR_t_),
+      EtaA_t(EtaA_t_),
+      Zeta_t(Zeta_t_),
+      ShfA_t(ShfA_t_),
+      ShfZ_t(ShfZ_t_) {
+  radial_length = radial_sublength * num_species;
+  angular_length = angular_sublength * (num_species * (num_species + 1) / 2);
+}
+
+Result::Result(
+    Tensor aev_t_,
+    Tensor tensor_Rij_,
+    Tensor tensor_radialRij_,
+    Tensor tensor_angularRij_,
+    int64_t total_natom_pairs_,
+    int64_t nRadialRij_,
+    int64_t nAngularRij_,
+    Tensor tensor_centralAtom_,
+    Tensor tensor_numPairsPerCenterAtom_,
+    Tensor tensor_centerAtomStartIdx_,
+    int64_t maxnbrs_per_atom_aligned_,
+    int64_t angular_length_aligned_,
+    int64_t ncenter_atoms_,
+    Tensor coordinates_t_,
+    Tensor species_t_)
+    : aev_t(aev_t_),
+      tensor_Rij(tensor_Rij_),
+      tensor_radialRij(tensor_radialRij_),
+      tensor_angularRij(tensor_angularRij_),
+      total_natom_pairs(total_natom_pairs_),
+      nRadialRij(nRadialRij_),
+      nAngularRij(nAngularRij_),
+      tensor_centralAtom(tensor_centralAtom_),
+      tensor_numPairsPerCenterAtom(tensor_numPairsPerCenterAtom_),
+      tensor_centerAtomStartIdx(tensor_centerAtomStartIdx_),
+      maxnbrs_per_atom_aligned(maxnbrs_per_atom_aligned_),
+      angular_length_aligned(angular_length_aligned_),
+      ncenter_atoms(ncenter_atoms_),
+      coordinates_t(coordinates_t_),
+      species_t(species_t_) {}
+
+Result::Result(tensor_list tensors)
+    : aev_t(tensors[0]),
+      tensor_Rij(tensors[1]),
+      tensor_radialRij(tensors[2]),
+      tensor_angularRij(tensors[3]),
+      total_natom_pairs(tensors[4].item<int>()),
+      nRadialRij(tensors[5].item<int>()),
+      nAngularRij(tensors[6].item<int>()),
+      tensor_centralAtom(tensors[7]),
+      tensor_numPairsPerCenterAtom(tensors[8]),
+      tensor_centerAtomStartIdx(tensors[9]),
+      maxnbrs_per_atom_aligned(tensors[10].item<int>()),
+      angular_length_aligned(tensors[11].item<int>()),
+      ncenter_atoms(tensors[12].item<int>()),
+      coordinates_t(tensors[13]),
+      species_t(tensors[14]) {}
+
+tensor_list Result::to_list() {
+  return {aev_t,
```

Comment:
should we remove aev from here, and in constructor `Result::Result(tensor_list tensors)`, just initialize `aev_t` to undefined tensor?